scripts/ci/check_compliance.py

```diff
@@ -337,9 +337,15 @@ def get_kconfig_dts(self, kconfig_dts_file):
         # not a module nor a pip-installed Python utility
         zephyr_drv_kconfig_path = os.path.join(ZEPHYR_BASE, "scripts", "dts",
                                                "gen_driver_kconfig_dts.py")
-        binding_path = os.path.join(ZEPHYR_BASE, "dts", "bindings")
+        binding_paths = []
+        binding_paths.append(os.path.join(ZEPHYR_BASE, "dts", "bindings"))
+        if ZEPHYR_BASE != GIT_TOP:
+            binding_paths.append(os.path.join(GIT_TOP, "dts", "bindings"))
```

Comment:
This will only add the bindings in the repo where you are running compliance. What about bindings in other repos included in the manifest?
I think we should do what we are doing with Kconfig instead:
https://github.com/zephyrproject-rtos/zephyr/blob/754f7ac1fb4a68538842b9de8492b09d4c193674/scripts/ci/check_compliance.py#L289

Reply:
as @carlescufi points out, we should cover all modules, else we just end here again later.

You can probably extend the `get_modules()` function to support `settings_file` as argument.
And then pass `--settings-out` as argument here:
https://github.com/zephyrproject-rtos/zephyr/blob/72c6ffb2d04bf1d9c23e0ef71464b3693d96ba98/scripts/ci/check_compliance.py#L307
```
'--settings-out', settings_file
```
and then pass the `settings_file` that was created.
For DTS, it will contain lines as:
```
"DTS_ROOT":"<path>"
```
other settings in the file can be ignored.

Long term, compliance will benefit from an overhaul, where build system gets an improved interface, so things are easier shared / reused between CMake and compliance.